festim/hydrogen_transport_problem.py

```diff
@@ -108,8 +118,37 @@ def temperature(self):
     def temperature(self, value):
         if value is None:
             self._temperature = value
+        elif isinstance(value, (float, int, fem.Constant, fem.Function)):
+            self._temperature = value
+        elif callable(value):
+            self._temperature = value
+        else:
+            raise TypeError(
+                f"Value must be a float, int, fem.Constant, fem.Function, or callable"
+            )
+
+    @property
+    def temperature_fenics(self):
+        return self._temperature_fenics
+
+    @temperature_fenics.setter
+    def temperature_fenics(self, value):
+        if value is None:
+            self._temperature_fenics = value
+            return
+        self._temperature_fenics = value
```

Comment:
Ok. I see that in this case, having a setter is kinda useless

The point of having a setter is to do some type checking.

Here, we want to make sure that `temperature_fenics` is either a `Constant` or a `Function` and raise an error otherwise